lnwallet/channel_test.go

```diff
@@ -833,6 +836,85 @@ func TestCheckHTLCNumberConstraint(t *testing.T) {
 
 }
 
+// TestCheckDustLimit checks that unsettled HTLC with dust limit not included in
+// commitment transaction as output, but sender balance is decreased (thereby all
+// unsettled dust HTLCs will go to miners fee).
+func TestCheckDustLimit(t *testing.T) {
+	createHTLC := func(data, amount btcutil.Amount) (*lnwire.HTLCAddRequest,
+		[32]byte) {
+		preimage := bytes.Repeat([]byte{byte(data)}, 32)
+		paymentHash := fastsha256.Sum256(preimage)
+
+		var returnPreimage [32]byte
+		copy(returnPreimage[:], preimage)
+
+		return &lnwire.HTLCAddRequest{
+			RedemptionHashes: [][32]byte{paymentHash},
+			Amount:           amount,
+			Expiry:           uint32(5),
+		}, returnPreimage
+	}
+
+	// Create a test channel which will be used for the duration of this
+	// unittest. The channel will be funded evenly with Alice having 5 BTC,
+	// and Bob having 5 BTC.
+	aliceChannel, bobChannel, cleanUp, err := createTestChannels(3)
+	if err != nil {
+		t.Fatalf("unable to create test channels: %v", err)
+	}
+	defer cleanUp()
+
+	htlcAmount := btcutil.Amount(1)
+	channelAmount := btcutil.Amount(5 * 1e8)
+
+	htlc, preimage := createHTLC(0, htlcAmount)
+	if _, err := aliceChannel.AddHTLC(htlc); err != nil {
+		t.Fatalf("alice unable to add htlc: %v", err)
+	}
+	if _, err := bobChannel.ReceiveHTLC(htlc); err != nil {
+		t.Fatalf("bob unable to receive htlc: %v", err)
+	}
+	forceStateTransition(aliceChannel, bobChannel)
+
+	// First two outputs are payment to them and to us. If we encounter
+	// third output it means that dust HTLC was included. Their channel
+	// balance shouldn't change because, it will be changed only after dust
+	// HTLC will be settled.
+	commitment := aliceChannel.localCommitChain.tip()
```

Comment:
I have a branch locally which starts to factor out paramter like this into a `FundingConfig` struct which includes all the parameters as are currently defined within the spec.